dataset/src/main/kotlin/org/jetbrains/kotlinx/dl/dataset/preprocessor/image/Resize.kt

```diff
@@ -33,7 +33,12 @@ public class Resize(
     public var renderingSpeed: RenderingSpeed = RenderingSpeed.MEDIUM,
     public var enableAntialiasing: Boolean = true
 ) : ImagePreprocessor {
-    override fun apply(image: BufferedImage, inputShape: ImageShape): Pair<BufferedImage, ImageShape> {
+
+    override fun getOutputShape(inputShape: ImageShape?): ImageShape {
+        return ImageShape(outputWidth.toLong(), outputHeight.toLong(), 3)
```

Comment:
Could it be applied to the 1-channel images?

Reply:
I think so, however the output will have 3 channels anyway. Ideally we'd want to keep the number of channels in the image. So `Resize#getOutputShape` should return `ImageShape(outputWidth.toLong(), outputHeight.toLong(), null)` (we need to make `ImageShape#channels` nullable for that).